src/OpenIDConnectClient.php

```diff
@@ -1087,6 +1087,18 @@ private function verifyRSAJWTSignature(string $hashType, stdClass $key, $payload
         return $key->verify($payload, $signature);
     }
 
+    private function verifyEdDSAJWTsignature($key, $payload, $signature) {
+        if (!(property_exists($key, 'x'))) {
+            throw new OpenIDConnectClientException('Malformed key object');
+        }
+
+        if (!function_exists("sodium_crypto_sign_verify_detached")) {
```

Comment:
This can be handled via composer as well. Please adjust composer.json and add 
`        "ext-sodium": "*",`